ci/run-example.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ -z "${PROJECT_ROOT+x}" ]; then
+  readonly PROJECT_ROOT="$(cd "$(dirname $0)/.."; pwd)"
+fi
+source "${PROJECT_ROOT}/ci/colors.sh"
+
+# If an example fails, this is set to 1 and the program exits with failure.
+EXIT_STATUS=0
+
+# The name of the emulator / testbench log file
+EMULATOR_LOG=""
```

Comment:
After jumping between files in this review, it seems confusing to me to assign a default value to these globals here -- the value of each is then dependent on when we `source` this script (i.e. if we source it *after* setting this value, it erases whatever was previously set).  Can you make this a conditional that only sets the value to empty if no value currently exists? E.g. `EMULATOR_LOG="${EMULATOR_LOG:=}"`

...also/alternatively, maybe change the name of this script? It doesn't actually run an example; it initializes some global values and defines a function.  Maybe `init-example-runner`?  That would eliminate potential confusion about when this should be sourced, and eliminate my concern above :)